HiP-UserStore/Core/UserIndex.cs

```diff
@@ -58,27 +60,54 @@ public void ApplyEvent(IEvent e)
         {
             switch (e)
             {
-                case UserCreated ev:
+                case UserPhotoDeleted ev:
                     lock (_lockObject)
                     {
-                        _users.Add(ev.Id, new UserInfo { Email = ev.Properties?.Email });
-                        _auth0UserIds.Add(ev.UserId, ev.Id);
+                        if (_users.TryGetValue(ev.Id, out var info))
+                            info.ProfilePicturePath = null;
                     }
                     break;
 
-                case UserPhotoUploaded ev:
-                    lock (_lockObject)
+                case CreatedEvent ev:
+                    var resourceType = ev.GetEntityType();
+                    if (resourceType == ResourceTypes.User)
                     {
-                        if (_users.TryGetValue(ev.Id, out var info))
-                            info.ProfilePicturePath = ev.Path;
+                        lock (_lockObject)
+                        {
+                            _users.Add(ev.Id, new UserInfo());
+                        }
                     }
                     break;
 
-                case UserPhotoDeleted ev:
-                    lock (_lockObject)
+                case PropertyChangedEvent ev:
+                    resourceType = ev.GetEntityType();
+                    if (resourceType == ResourceTypes.User)
                     {
-                        if (_users.TryGetValue(ev.Id, out var info))
-                            info.ProfilePicturePath = null;
+                        switch (ev.PropertyName)
+                        {
+                            case "Email":
```

Comment:
It would be better if you would use the `nameof` operator here. The same goes for the following cases